src/util/LogSlowExecution.h

```diff
@@ -9,26 +9,46 @@
 
 class LogSlowExecution
 {
-
-    std::chrono::system_clock::time_point mStart;
-    std::string mName;
-    int mThresholdInMs;
-
   public:
-    LogSlowExecution(std::string eventName, int ms = 1000)
+    enum class Mode
+    {
+        EXECUTION,
+        START
```

Comment:
I think this enum is confusing: `LogSlowExecution` does not know that time before start will be measured, and `checkExecuted` can be called anywhere. I think it should be more generic, something like this:
```
enum class Mode 
{
    AUTOMATIC_RAII,
    MANUAL // In this mode, it is the caller's responsibility to check elapsed time
}
```